### ISSUE TITLE
Calling static function from within a class generates a warning

### ISSUE THREAD

**Issue by adamscoble** [godotengine/godot#69282]
### Godot version

4.0.beta6

### System information

Windows 10

### Issue description

Hi all,

Calling a static function from within the class that owns it generates a warning. I would expect that calling the function from within the class shouldn't generate the warning (particularly if that class doesn't have a class name).

Thanks!

### Steps to reproduce

Add the below to a script and note the Error (warning) it generates:

```
func _ready() -> void:
    a_static_func()

static func a_static_func() -> void:
    pass
```

### Minimal reproduction project

N/A

**Comment by clayjohn** [comment]
This happens because ``_ready()`` is not a static function. As the warning indicates, you are calling a static function directly on an instance of the class (the current running instance of the script).

So, in this case the warning is _technically_ correct. But clearly it is annoying, and you are not the first person to think so. 

The reason we added this warning was mostly because many functions that you used to call on class instances in 3.x were made static. Instead of modifying an object, they return a new object. For example:
```
var tex = ImageTexture.new()
tex.create_from_image(some_img)
```
Is now written like:
```
var tex = ImageTexture.create_from_image(some_img)
```

The warning is useful here as it indicates to the user that ``tex.create_from_image()`` is not modifying the ``tex`` resource.

Additionally, from a code-quality standpoint, it is good practice to always call static functions from the class so that future readers of the code can distinguish between a function being called on an instance of a class or the class itself. 

However, the situation you point out above is a bit tricky. Since there is no specific class name, you have no choice but to just call the function directly. Even worse, the error message tells you to call the function on the class that your script inherits from. In other words, if your script looks like:
```
extends Control

func _ready() -> void:
    a_static_func()

static func a_static_func() -> void:
    pass
```

Then the warning tells you to use ``Control.a_static_func()`` which is not only annoying, but clearly wrong. 

I think we can carve out an exception to the warning, so it doesn't apply in this case. If not, we may just have to disable the warning whenever a static function is called from a method of the class it is defined in.

**Comment by adamscoble** [comment]
Thanks for the reply @clayjohn! 

I completely understand (and agree with) the purpose of the warning. From my perspective I think it makes sense to be able to call a static function from its class instance (particularly because some classes might not names as in your example). If you have a specific exception for unnamed classes then you create two different rules (and adding a class name later would mean refactoring any calls to it in the class). But I'll just follow Godot's lead! Haha.

This does beg a different question though: should I be naming more/all classes? I tend to limit it to things that need to be referenced elsewhere without a hierarchy relationship. Is the future of Godot every class being named? Or is my current philosophy correct?

**Comment by Zylann** [comment]
To give some reference, in C# the code will not compile if you access a static method with `instance.StaticMethod()`. But it will compile happily if you call `StaticMethod` from a non-static method of `instance`. https://ideone.com/rLmBHF

**Comment by dalexeev** [comment]
* Fixed by #85918.
